### PR TITLE
Tab completion

### PR DESCRIPTION
**Pull request by BloodAxe**
See documentation file in this PR for more details

### CODE REVIEW

**Comment by Louis-Dupont** [comment]
Just follow the instructions and it ran pretty fast on my computer (Mac), but I only see some keys. Is it normal?
image

**Comment by BloodAxe** [comment]
I would say yes, it does not show you "all" keys for a given config. For some reason root settings are not there. Perhaps we should evaluate this feature more thoroughly to see whether it is really helps.

**Comment by BloodAxe** [comment]
It is. I assume you expected to see all root parameters, right? We have no control on tab completion so if it is not providing useful completion - i guess we can reconsider whether we want to have this feature.

File: documentation/source/TabCompletion.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+compinit
+autoload -Uz bashcompinit && bashcompinit
+```
+
+See [Tab Completion](https://hydra.cc/docs/tutorials/basic/running_your_app/tab_completion/) for more information.
+
+## Caveats & Limitations
+
+* Tab completion is not supported on Windows.
+* You need to enable tab completion explicitly for each entry point in your application. In other words, if your project.
```
Comment (shaydeci) [inline documentation/source/TabCompletion.md:25]: if your project....
Line swapped here I believe.

File: documentation/source/TabCompletion.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+* Tab completion is not supported for running train_from_recipe using `-m` form: `python -m super_gradients.train_from_recipe ...`.
```
Comment (shaydeci) [inline documentation/source/TabCompletion.md:26]: Why not actually?
Cant we just do the same wherever SG package is stored in ? Or is it a problem with the fact that its being ran as a module?

**Review by shaydeci — CHANGES_REQUESTED** [review]

**Comment by shaydeci** [comment]
Im closing this down, as I dont think we currently benefit enough from tab completion.
Perhaps we can revisit this in the future, when tab completion is more ready from Hydra's side.

**Pull request closed without merging**